### ISSUE TITLE
[unity] Editor throws on startup (Unity 2020.2 beta 4+)

### ISSUE THREAD

**Issue by Nirlah** [EsotericSoftware/spine-runtimes#1779]
Unity: 2020.2 beta 4 and above
Spine runtime: Tested both with `2020-09-29` and `2020-10-07` versions

When using Spine runtime with 2020.2 beta the Editor throws `ReleaseAllScriptCaches did not release all script caches!` on startup.

It works fine with `2020.2 beta 2`.

Repro with Unity `2020.2b6` and `Spine Runtime 2020-10-07`: [repro.zip](https://github.com/EsotericSoftware/spine-runtimes/files/5351942/Archive.zip)

**Comment by HaraldCsaszar** [comment]
Thanks for reporting! Please note that we don't support Unity 2020.2 beta or any beta updates in general, as it may change a lot until released. We are happy to fix this issue though if it persists when 2020.2 is out of beta. Then we will gladly reopen the ticket.

**Comment by Nirlah** [comment]
Hey @HaraldCsaszar, Unity 2020.2 is out...
The error is still thrown on startup.

If it helps pinpoint the issue, it only happens on project startup, but not after scripts are recompiled (for example, switching to/from debug mode or changing scripts).

It has some critical implications for our projects; for example, materials aren't loaded until we explicitly trigger scripts recompile and reload.

**Comment by HaraldCsaszar** [comment]
Thanks for the info! I have reopened the ticket.

As always, please do not use Unity versions that are not listed as compatible with spine-unity in production.

**Comment by HaraldCsaszar** [comment]
Probably a related error has been posted on the forum:
> "No script asset for SpinePreferences. Check that the definition is in a file of the same name and that it compiles properly."

http://esotericsoftware.com/forum/Unity-2020-2f1-15128?p=66489#p66489
The forum posting contains reproduction steps.

**Comment by omgnuts** [comment]
Seem to be related to URP. Able to reproduce even when I manually installed URP on a clean 2020.2 by following these steps in this link here: https://docs.unity3d.com/Packages/com.unity.render-pipelines.universal@7.1/manual/InstallURPIntoAProject.html

TLDR, on the manual install:
- Setup a clean 2020.2 without URP, and with Spine working.
- Import Universal RP via Window > Package Manager
- Right-click on Asset in the Project window, and select Create > Rendering > Universal Render Pipeline > Pipeline Asset (Forward Renderer)

Warning: No script asset for SpinePreferences. Check that the definition is in a file of the same name and that it compiles properly
Error: ReleaseAllScriptCaches did not release all script caches! 

Finally, turning off URP doesn't stop the error. Had to recreate a whole new project (w/o URP) & brought my assets into the new workspace. 

Thanks!

**Comment by Nirlah** [comment]
@HaraldCsaszar 
I can confirm that the following solution successfully solves the issue: http://esotericsoftware.com/forum/Unity-2020-2f1-15128?sid=7a038ea00bb0e008bb3c4b1994829b0a&p=66806#p66806

**Comment by HaraldCsaszar** [comment]
Thanks very much for the confirmation @Nirlah! We have just submitted a fix based on this implementation. New 3.8 unitypackages are available for download, 4.0-beta unitypackage should follow in a few days.
